docs/_static/version-switcher/versions.json

```diff
@@ -1,118 +0,0 @@
-[
```

Comment:
So let's talk about what happens for patch releases, as said in your description: I think anytime we do a release, the version switcher should be updated to include it. And we probably shouldn't retire releases from docs as quickly as we will stop maintaining them. IMO, we should consider populating the list from the list of available versions in S3 plus the one currently being built. This requires checking right before publishing, so that if we do a couple of releases in short-ish succession, we don't overwrite one of them because it's docs weren't actually published yet, and the build process of the other release didn't know about it. And then we would just need to decide when (if ever) we retire a release from docs.

What are your thoughts?

Reply:
I'm not sure I'm familiar enough with the docs publishing part of the automation to answer reliably as part of this review. What exactly would we be reading from S3? With this current changeset, though, every new minor or patch release will get a corresponding docs version published, which includes appending to the version picker `versions.json` file. This all happens without relying on external services, because the git tags should be the source of truth for which versions we've published. But there's extra docs automation happening that I didn't look at as part of this work (mostly because it never came up), and that might require alteration, ideally as a separate pull request. If we want to pick and choose versions for docs, I would argue that we should then decouple the docs entirely from the Determined release process.

I'm also not sure if the existing documentation automation will handle jumping around between versions well (e.g. if we do a patch release on an older, minor release branch), because of the assumptions that code still makes. I'd need to spend more time looking closely; the docs automation is doing a bunch of stuff that I haven't needed to look at yet, and I don't think any of those necessary changes would make sense to go in as part of this PR.